tools/ete/ete_gene_cnv.py

```diff
@@ -0,0 +1,86 @@
+from __future__ import print_function
+
+import argparse
+import collections
+
+from ete3 import PhyloTree
+
+
+def printCSV(myDict, speciesList, colList=None):
+    """ Pretty print a list of dictionaries (myDict) as a dynamically sized table.
+    If column names (colList) aren't specified, they will show in random order.
+    Author: Thierry Husson - Use it as you want but don't blame me.
+    """
+    colList = ["gene"]
+    colList.extend(speciesList)
+
+    if not colList:
+        colList = list(myDict[0].keys() if myDict else [])
+
+    myList = [colList]
+
+    for item in myDict:
+        myList.append([str(item[col] if item[col] is not None else '') for col in colList])
+
+    for item in myList:
+        print(*item, sep=", ")
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Gene Copy Number Finder')
+    parser.add_argument('--genetree', help='GeneTree in nhx format')
+    parser.add_argument('--speciesorder', help='Comma separated species list')
+    args = parser.parse_args()
+
+    if args.genetree is None:
+        parser.error("--genetree option must be specified, GeneTree in nhx format")
+
+    if args.speciesorder is None:
+        parser.error("--speciesorder option must be specified, Comma separated species list")
+
+    f = open(args.genetree, "r")
+
+    speciesList = args.speciesorder.split(",")
+
+    table = []
+
+    # reads multiple gene tree line by line gene tree
+    for row in f:
+
+        contents = row
+
+        # Remove empty NHX features that can be produced by TreeBest but break ete3
+        contents = contents.replace('[&&NHX]', '')
+
+        # reads single gene tree
+        genetree = PhyloTree(contents)
+
+        leaves = genetree.get_leaf_names()
+
+        # Genetree nodes are required to be in gene_species format
+        leaves_list = [_ for _ in leaves if '_' in _]
```

Comment:
Should we raise an issue if a leaf node doesn't contain `_` or just filter it out (as done here)?